### PR TITLE
Allow to disable --recursive fetch

### PR DESCRIPTION
**Pull request by mfojtik**
This allows you to set `--recursive=false` to disable `git clone --recursive`.

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
@soltysh || @jhadvig review pls?

**Comment by soltysh** [comment]
LGTM

**Comment by jhadvig** [comment]
@mfojtik LGTM :+1:

**Comment by mfojtik** [comment]
(wait for Jenkins to finish, I have bad feelings about integration tests)

**Comment by soltysh** [comment]
[merge]

File: cmd/s2i/main.go [hunk 1]
```diff
@@ -187,6 +187,7 @@ func newCmdBuild(cfg *api.Config) *cobra.Command {
 	buildCmd.Flags().BoolVarP(&(cfg.Quiet), "quiet", "q", false, "Operate quietly. Suppress all non-error output.")
 	buildCmd.Flags().BoolVar(&(cfg.RunImage), "run", false, "Run resulting image as part of invocation of this command.  You may have to Ctrl-C to exit the sti command invocation and the container launched because this options was specified.")
 	buildCmd.Flags().BoolVar(&(cfg.Incremental), "incremental", false, "Perform an incremental build")
+	buildCmd.Flags().BoolVar(&(cfg.DisableRecursive), "recursive", true, "Fetch all git submodules when cloning application repository")
```
Comment (bparees) [inline cmd/s2i/main.go]: @mfojtik @soltysh @jhadvig why did we call the flag recursive and the env variable DisableRecursive?  wtf?
see: https://github.com/openshift/source-to-image/issues/530

Comment (soltysh) [inline cmd/s2i/main.go]: OMG, I'm hiding as deep as possible. I can only offer the fix in that case :innocent:

Comment (soltysh) [inline cmd/s2i/main.go]: And I just noticed you've already handled that one, I owe you one another... :wink:

**Pull request merged**